compiler/include/stmt.h

```diff
@@ -275,6 +276,46 @@ class GotoStmt : public Stmt {
   LabelSymbol*        gotoTarget()                                     const;
 };
 
+
+/************************************ | *************************************
+*                                                                           *
+*                                                                           *
+************************************* | ************************************/
+
+//
+// forall loop statement
+//
+class ForallStmt : public Stmt
+{
+public:
+  // All pointer fields are always non-null.
+  BlockStmt*     fsBlock;   // contains defs for index variable(s) and fBody
+  AList          fIter;     // iterable expression(s)
+  bool           fZippered;
+  ForallIntents* fWith;     // todo: replace with AList of per-variable "clauses"
+
+  // derived from the above
+  VarSymbol* fIdxVar()     const;
+  VarSymbol* fIdxCopyVar() const;
+  BlockStmt* fBody()       const; // forall loop body
```

Comment:
For other work I am doing (ref-if-modified), it would be really useful to be able to go from index variable to iterator variable. I wonder if there should be a method to access that information?

I also find it a little odd that these methods start with 'f'. I'd expect the ForallStmt to have fields internal to the implementation, such as _block, _iter, _zippered, _with; and then to have methods with names like indexVariable() etc. to access them.

Also, it seems to me that we'd like to have ways of accessing particular index variables or iterators when doing zippered iteration. I wonder if we should have:

 * get number of zippered iterables
 * get index variable i
 * get iterable expression i